model/src/channel/message/kind.rs

```diff
@@ -1,43 +1,44 @@
-use crate::channel::ConversionError;
-use serde_repr::{Deserialize_repr, Serialize_repr};
+use serde::{Deserialize, Serialize};
 
 #[derive(
-    Clone, Copy, Debug, Deserialize_repr, Eq, Hash, Ord, PartialEq, PartialOrd, Serialize_repr,
+    Clone, Copy, Debug, Deserialize, Eq, Hash, Ord, PartialEq, PartialOrd, Serialize,
 )]
-#[repr(u8)]
+#[serde(from = "u8", into = "u8")]
 pub enum MessageType {
-    Regular = 0,
-    RecipientAdd = 1,
-    RecipientRemove = 2,
-    Call = 3,
-    ChannelNameChange = 4,
-    ChannelIconChange = 5,
-    ChannelMessagePinned = 6,
-    GuildMemberJoin = 7,
-    UserPremiumSub = 8,
-    UserPremiumSubTier1 = 9,
-    UserPremiumSubTier2 = 10,
-    UserPremiumSubTier3 = 11,
-    ChannelFollowAdd = 12,
-    GuildDiscoveryDisqualified = 14,
-    GuildDiscoveryRequalified = 15,
-    GuildDiscoveryGracePeriodInitialWarning = 16,
-    GuildDiscoveryGracePeriodFinalWarning = 17,
-    ThreadCreated = 18,
+    Regular,
+    RecipientAdd,
+    RecipientRemove,
+    Call,
+    ChannelNameChange,
+    ChannelIconChange,
+    ChannelMessagePinned,
+    GuildMemberJoin,
+    UserPremiumSub,
+    UserPremiumSubTier1,
+    UserPremiumSubTier2 ,
+    UserPremiumSubTier3 ,
+    ChannelFollowAdd ,
+    GuildDiscoveryDisqualified ,
+    GuildDiscoveryRequalified ,
+    GuildDiscoveryGracePeriodInitialWarning ,
+    GuildDiscoveryGracePeriodFinalWarning ,
+    ThreadCreated ,
     /// Message is an inline reply.
-    Reply = 19,
+    Reply,
     /// Message is a chat input command.
-    ChatInputCommand = 20,
-    ThreadStarterMessage = 21,
-    GuildInviteReminder = 22,
-    ContextMenuCommand = 23,
+    ChatInputCommand,
+    ThreadStarterMessage,
+    GuildInviteReminder,
+    ContextMenuCommand,
+
+    /// Message is of a type not (yet) known to this library
```

Comment:
Yeah, referring to my other comment about documentation these we should definitely pick a standard